frontend/src/components/AddLoan.vue

```diff
@@ -61,91 +61,94 @@
 </template>
 
 <script lang="ts">
-import { Component, Watch, Prop, Vue } from 'vue-property-decorator';
 import LoansModule from '../store/modules/LoansModule';
 import { Loan, AddLoanForm } from '../types';
 import UsersModule from '../store/modules/UsersModule';
+import { defineComponent, ref, SetupContext, watch } from '@vue/composition-api';
+export default defineComponent({
+  name: 'AddLoan',
+  setup(_ : object, { root, emit } : SetupContext ) {
+    
+    let inputText = ref(''),
+      inputState = ref(false),
+      inputReturn = ref(''),
+      button = ref(null),
+      input = ref(''),
+      menu = ref(false as boolean);
 
-@Component
-export default class AddLoan extends Vue {
-  public input: string = '';
-  public inputState: boolean | null = null;
-  public menu: boolean = false;
-  public inputReturn: string = '';
-
-  public form: AddLoanForm  = {
-    book_id: 0, //eslint-disable-line camelcase
-    /* Hardcoded value atm for the school */
-    lent_by_id: 1854282603, //eslint-disable-line camelcase
-    loaned_by_id: 0,//eslint-disable-line camelcase
-    expiration_date: '2020/02/12' //eslint-disable-line camelcase
-  };
-
+    const form: AddLoanForm  = {
+      book_id: 0, //eslint-disable-line camelcase
+      /* Hardcoded value atm for the school */
+      lent_by_id: 1854282603, //eslint-disable-line camelcase
+      loaned_by_id: 0,//eslint-disable-line camelcase
+      expiration_date: '2020/02/12' //eslint-disable-line camelcase
+    };
 
-  public onCancel(): void {
-    this.menu = false;
-  }
-
-  public onOk(): void {
-    if (!this.input) {
-      this.inputState = false;
+    function onCancel(): void {
+      menu.value = false;
     }
-    if (this.input) {
-      this.onCancel();
-      /* "Return" our popover "form" results */
-      this.inputReturn = this.input;
-      this.form.book_id = parseInt(this.input); //eslint-disable-line camelcase
-      this.form.loaned_by_id = UsersModule.currentUserID; //eslint-disable-line camelcase
-      console.log(this.form);
-      /* This Checks if the form contains the necessary data  */
-      if (!!this.form.lent_by_id && !!this.form.loaned_by_id && !!this.form.book_id) {
-        LoansModule.create(this.form)
-          .then((payload: Loan) => this.$emit('loan-added', payload))
-          .catch((failure: boolean) => console.log(failure));
-        this.input = '';
+
+    function onOk() : void{
+      if(!inputText.value){
+        inputState.value = false;
+      }
+      if (!input.value) {
+        onCancel();
+        /* "Return" our popover "form" results */
+        inputReturn.value = input.value;
+        form.book_id = parseInt(input.value); //eslint-disable-line camelcase
+        form.loaned_by_id = UsersModule.currentUserID; //eslint-disable-line camelcase
+        console.log(form);
+        /* This Checks if the form contains the necessary data  */
+        if (!!form.lent_by_id && !!form.loaned_by_id && !!form.book_id) {
```

Comment:
```suggestion
        if (!Object.values(form.value).some(prop => prop === 0)) {
```
Somewhat more clean solution.